sdcm/utils/decorators.py

```diff
@@ -170,23 +170,40 @@ def latency_calculator_decorator(original_function: Optional[Callable] = None, *
     def wrapper(func):
 
         @wraps(func)
-        def wrapped(*args, **kwargs):  # pylint: disable=too-many-branches, too-many-locals
+        def wrapped(*args, **kwargs):  # noqa: PLR0914
+            from sdcm.tester import ClusterTester
+            from sdcm.nemesis import Nemesis
             start = time.time()
-            start_node_list = args[0].cluster.nodes[:]
-            reactor_stall_stats = {}
+            # If the decorator is applied dynamically, "self" argument is not transferred  via "args" and may be found in bounded function
+            _self = getattr(func, "__self__", None) or args[0]
+            if isinstance(_self, ClusterTester):
```

Comment:
We can check the `_self.__class__` value instead of making unnecessary imports:

```
_self = getattr(func, "__self__", None) or args[0]

if "ClusterTester" in _self.__class__:
    ...
elif "Nemesis" in _self.__class__:
    ...
else:
 raise ValueError(...)
```

Reply:
I rather not, that won't work for inheritage